xc/device.ml

```diff
@@ -1415,18 +1415,31 @@ module Vusb = struct
   let exec_usb_reset_script argv =
     try
       let stdout, stderr = Forkhelpers.execute_command_get_output !Xc_resources.usb_reset_script argv in
-      debug "usb_reset script %s returned stdout=%s stderr=%s" (String.concat " " argv) stdout stderr
+      debug "usb_reset script %s returned stdout=%s stderr=%s" (String.concat " " argv) stdout stderr;
+      stdout
     with err ->
       error "Failed to call usb_reset script with arguments %s" (String.concat " " argv);
       raise (Internal_error (Printf.sprintf "Call to usb reset failed: %s" (Printexc.to_string err)))
 
   let cleanup domid =
     try
       let cmd = ["cleanup"; "-d"; string_of_int domid] in
-      exec_usb_reset_script cmd
+      ignore(exec_usb_reset_script cmd)
     with err ->
       warn "Failed to clean up VM %s: %s" (string_of_int domid) (Printexc.to_string err)
 
+  let save_uid_gid ~xs ~id ~domid ~uid_gid =
+    (* Save the uid gid of the device file to xenstore, as we need to restore
+       the uid/gid when the vusb is unplgugged.
+       eg. /local/domain/22/usb/vusb1-3-2-3/uid = "0" *)
+    let usb_path = Printf.sprintf "/local/domain/%d/usb/%s" domid
+        (Stdext.Xstringext.String.split '.' id |> String.concat "-") in
+    let dev_own= [
+      "uid",         List.nth uid_gid 0;
```

Comment:
`List.nth` can fail. It would be better to pattern match:
```
match uid_gid with
| [uid; gid] ->
| _  -> error handling
```

Reply:
Yes. For the approach, I also have some concern. If we can store it someplace, it will be a better approach. @jonludlam, do you have any suggestions about this? Thanks.

Reply:
And we can read from file system directly but the script already gets the uid/gid by parsing the related file which I discussed with @TalonsLee about this. We think we don't do it again.